app/services/ocrService.py

```
@@ -235,12 +222,22 @@ def save_pdrecord_json(data: Dict[str, Any], db: Session) -> Record:
+        db.commit()
+    except IntegrityError as e:
+        db.rollback()
+        # record_date 중복인 경우에만 명확한 메시지 제공
+        if "record_date" in str(e.orig):
+            detail = f"{record_date.isoformat()} 해당 기록이 이미 존재합니다."
+        else:
+            detail = "데이터 무결성 제약 조건을 위반했습니다."
+    raise HTTPException(
+        status_code=409,
+        detail = detail,
+    ) from e
```

Ruff (0.14.3)

Undefined name `e`

(F821)